plugins/vpp/ifplugin/vppcalls/dump_interface_vppcalls.go

```diff
@@ -113,14 +125,26 @@ func (handler *ifVppHandler) DumpInterfaces() (map[uint32]*InterfaceDetails, err
 		}
 	}
 
-	// Get vrf for every interface
+	// Get DHCP clients
+	dhcpClients, err := handler.dumpDhcpClients()
+	if err != nil {
+		return nil, fmt.Errorf("failed to dump interface DHCP clients: %v", err)
+	}
+	// Get vrf for every interface and fill DHCP if set
 	for _, ifData := range ifs {
+		// VRF
 		vrf, err := handler.GetInterfaceVRF(ifData.Meta.SwIfIndex)
 		if err != nil {
 			handler.log.Warnf("Interface dump: failed to get VRF from interface %d: %v", ifData.Meta.SwIfIndex, err)
 			continue
 		}
 		ifData.Interface.Vrf = vrf
+		// DHCP
+		dhcpData, ok := dhcpClients[ifData.Meta.SwIfIndex]
+		if ok {
+			ifData.Interface.SetDhcpClient = true
```

Comment:
What does `SetDhcpClient` client mean, and is it enough to find client swifindex in dhcp clients dump to set this to true? What about the state of dhcp lease? 

Asking just for clarification.

Reply:
An interface only needs to be set as a DHCP client using its sw_if_index (API dhcp_client_config). Dump returns only interfaces set as DHCP, so yet it is enough.

I will add DHCP lease to the metadata